spall.h

```diff
@@ -250,70 +261,101 @@ static bool Spall__BufferWrite(SpallProfile *ctx, SpallBuffer *wb, void *p, size
     return true;
 }
 
-bool SpallBufferFlush(SpallProfile *ctx, SpallBuffer *wb) {
+bool spall_buffer_flush(SpallProfile *ctx, SpallBuffer *wb) {
 #ifdef SPALL_DEBUG
     if (!wb) return false;
     if (!wb->data) return false;
 #endif
 
-    if (!Spall__BufferFlush(ctx, wb)) return false;
+    if (!spall__buffer_flush(ctx, wb)) return false;
     return true;
 }
 
-bool SpallBufferInit(SpallProfile *ctx, SpallBuffer *wb) {
-    if (!SpallBufferFlush(NULL, wb)) return false;
+bool spall_buffer_init(SpallProfile *ctx, SpallBuffer *wb) {
+    if (!spall_buffer_flush(NULL, wb)) return false;
     wb->ctx = ctx;
     return true;
 }
-bool SpallBufferQuit(SpallProfile *ctx, SpallBuffer *wb) {
-    if (!SpallBufferFlush(ctx, wb)) return false;
+bool spall_buffer_quit(SpallProfile *ctx, SpallBuffer *wb) {
+    if (!spall_buffer_flush(ctx, wb)) return false;
     wb->ctx = NULL;
     return true;
 }
 
-bool SpallBufferAbort(SpallBuffer *wb) {
+bool spall_buffer_abort(SpallBuffer *wb) {
     if (!wb) return false;
     wb->ctx = NULL;
-    if (!Spall__BufferFlush(NULL, wb)) return false;
+    if (!spall__buffer_flush(NULL, wb)) return false;
     return true;
 }
 
-static SpallProfile Spall__Init(const char *filename, double timestamp_unit) {
+SpallHeader spall_build_header(double timestamp_unit) {
+    SpallHeader header;
+    header.magic_header = 0x0BADF00D;
+    header.version = 1;
+    header.timestamp_unit = timestamp_unit;
+    header.must_be_0 = 0;
+    return header;
+}
+SpallBeginEventMax spall_build_begin(const char *name, signed long name_len, const char *args, signed long args_len, double when, uint32_t tid, uint32_t pid, int *event_len) {
```

Comment:
Wondering if a common usage API would be:

```cpp
int max = 100000;
void *b = malloc(max);
int n = 0;
n += spall_build_begin(b + n, max - n);
n += spall_build_end(b + n, max - n);
```

Worth changing to facilitate this case?